packages/controls/src/color/color.ts

```diff
@@ -0,0 +1,118 @@
+import { match } from 'ts-pattern'
+import { z } from 'zod'
+
+import {
+  ControlDataTypeKey,
+  colorDataSchema,
+  type ColorData,
+  type CopyContext,
+} from '../common'
+import { controlTraitsRegistry } from '../registry'
+
+import {
+  type VersionedControlDefinition,
+  type ControlTraits,
+  type ParseResult,
+} from '../traits'
+
+export const Color = controlTraitsRegistry.add(
+  (() => {
+    const type = 'makeswift::controls::color' as const
+    const v1DataType = 'color::v1' as const
+    const version = 1 as const
+
+    const dataSignature = {
+      v1: { [ControlDataTypeKey]: v1DataType },
+    } as const
+
+    const dataSchema = z.union([
+      colorDataSchema,
+      colorDataSchema.and(
+        z.object({
+          [ControlDataTypeKey]: z.literal(v1DataType),
+        }),
+      ),
+    ])
+
+    const configSchema = z.object({
+      label: z.string().optional(),
+      labelOrientation: z
+        .union([z.literal('horizontal'), z.literal('vertical')])
+        .optional(),
+      defaultValue: z.string().optional(),
+      hideAlpha: z.boolean().optional(),
+    })
+
+    type ControlData = z.infer<typeof dataSchema>
+    type Config = z.infer<typeof configSchema>
+
+    type ControlDefinition<C extends Config = Config> =
+      VersionedControlDefinition<typeof type, C, ColorData, typeof version>
```

Comment:
Is there a reason for these to be defined inside the IIFE?

Reply:
That's the easiest way to limit the scope of these definitions. The main point of this refactoring is defining a common traits interface, and making all the type info and functions accessible through the interface rather through an ad-hoc collection of top-level types and functions.